docs: Use the generated Serverpod class in initialization examples (#759)
serverpod/serverpod_docs: docs/11-upgrading/01-upgrade-to-four.md

## What's new in 4.0

- **`serverpod start` TUI**: hot reload on save, **R** to hot restart, **M** to create a migration, **A** to apply migrations, **P** to apply a repair migration.

> - **Simplified server initialization**: the generated `Serverpod` class comes pre-wired with your project's `Protocol` and `Endpoints`, so `server.dart` creates the server with just `Serverpod(args)`. The classic `Serverpod(args, Protocol(), Endpoints())` form keeps working.

Reviewer: Suggestion: Make this a little less verbose to match the list style.

```suggestion
- **Simplified server initialization** with the generated `Serverpod` class that pre-wires the `Protocol` and `Endpoints`; optional, `Serverpod(args, Protocol(), Endpoints())` still work.
```

Reviewer 2: Agreed on the length. One thing to add: the two forms need different imports. The generated class takes only `args` as a positional argument and does not re-export `Protocol` or `Endpoints`, so the old call only works if you keep the old imports.

```suggestion
- **Simplified server initialization** with the generated `Serverpod` class that pre-wires `Protocol` and `Endpoints`, so `server.dart` needs only `Serverpod(args)`. Projects that keep their existing imports can stay on `Serverpod(args, Protocol(), Endpoints())`.
```

Author: Done! Went with @developerjamiu version since it also captures that the two forms need different imports.
